src/sentry/integrations/opsgenie/actions/form.py

```diff
@@ -0,0 +1,84 @@
+# mypy: ignore-errors
+from __future__ import annotations
+
+from typing import Any, Mapping
+
+from django import forms
+from django.utils.translation import gettext_lazy as _
+
+from sentry.services.hybrid_cloud.integration import integration_service
+from sentry.services.hybrid_cloud.integration.model import RpcOrganizationIntegration
+
+
+def _validate_int_field(field: str, cleaned_data: Mapping[str, Any]) -> int | None:
+    value_option = cleaned_data.get(field)
+    if value_option is None:
+        return None
+
+    try:
+        return int(value_option)
+    except ValueError:
+        raise forms.ValidationError(_(f"Invalid {field}"), code="invalid")
+
+
+class OpsgenieNotifyTeamForm(forms.Form):
+    """Used for notifying a specific team."""
+
+    account = forms.ChoiceField(choices=(), widget=forms.Select())
+    team = forms.ChoiceField(required=False, choices=(), widget=forms.Select())
+
+    # mypy is incorrectly flagging the fields as Field types instead of
+    # ChoiceFields, which is the reason for the top-level type : ignore-errors.
+    def __init__(self, *args, **kwargs):
+        integrations = [(i.id, i.name) for i in kwargs.pop("integrations")]
+        teams = kwargs.pop("teams")
+
+        super().__init__(*args, **kwargs)
+        if integrations:
+            self.fields["account"].initial = integrations[0][0]
+
+        self.fields["account"].choices = integrations
+        self.fields["account"].widget.choices = self.fields["account"].choices
+
+        if teams:
+            self.fields["team"].initial = teams[0][0]
+
+        self.fields["team"].choices = teams
+        self.fields["team"].widget.choices = self.fields["team"].choices
+
+    def _team_is_valid(
+        self, team_id: str | None, org_integrations: list[RpcOrganizationIntegration]
+    ) -> bool:
+        for oi in org_integrations:
+            teams = oi.config.get("team_table")
+            if not teams:
+                continue
+            for team in teams:
+                if team["id"] == team_id:
+                    return True
```

Comment:
I think you can shorten this a bit for legibility. We're just ensuring that the team is associated with an org_integration, and I believe this accomplishes that.
```python
if teams and any(team["id"] == team_id for team in teams):
  return True
```